plugins/json_importer.cc

```diff
@@ -0,0 +1,239 @@
+/* xlsx_importer.cc
+   Francois Maillet, 19 janvier 2016
+
+   This file is part of MLDB. Copyright 2016 Datacratic. All rights reserved.
+   
+   Importer for text files containing a JSON per line
+*/
+
+#include "mldb/core/procedure.h"
+#include "mldb/core/dataset.h"
+#include "mldb/types/value_description.h"
+#include "mldb/types/structure_description.h"
+#include "mldb/vfs/filter_streams.h"
+#include "mldb/types/any_impl.h"
+#include "mldb/plugins/for_each_line.h"
+#include <boost/lexical_cast.hpp>
+
+using namespace std;
+
+
+namespace Datacratic {
+namespace MLDB {
+
+
+/*****************************************************************************/
+/* JSON IMPORTER                                                             */
+/*****************************************************************************/
+
+struct JSONImporterConfig : ProcedureConfig {
+
+    JSONImporterConfig() :
+          limit(-1),
+          offset(0),
+          ignoreBadLines(false)
+    {}
+
+    Url dataFileUrl;
+    PolyConfigT<Dataset> output;
+    
+    int64_t limit;
+    int64_t offset;
+    bool ignoreBadLines;
+};
+
+DECLARE_STRUCTURE_DESCRIPTION(JSONImporterConfig);
+
+DEFINE_STRUCTURE_DESCRIPTION(JSONImporterConfig);
+
+JSONImporterConfigDescription::
+JSONImporterConfigDescription()
+{
+    addField("dataFileUrl", &JSONImporterConfig::dataFileUrl,
+             "URL to load text file from");
+    addField("output", &JSONImporterConfig::output,
+             "Configuration for output dataset",
+             PolyConfigT<Dataset>().withType("sparse.mutable"));
+    addField("limit", &JSONImporterConfig::limit,
+             "Maximum number of lines to process");
+    addField("offset", &JSONImporterConfig::offset,
+            "Skip the first n lines (excluding the header if present).", int64_t(0));
+    addField("ignoreBadLines", &JSONImporterConfig::ignoreBadLines,
+             "If true, any line causing an error will be skipped.", false);
```

Comment:
Are json objects with missing values considered an error?  Not sure if this option is well-defined for json.